Twilio/Security/RequestValidator.php

```diff
@@ -13,6 +13,31 @@ function __construct($authToken) {
     }
 
     public function computeSignature($url, $data = array()) {
+        // username and password from HTTP URLs and username, password and 
+        // port from HTTPS URLs must be dropped before the calculation,
+        // see https://www.twilio.com/docs/api/security#notes
+        $parsed_url = parse_url($url);
+        unset($parsed_url['user'], $parsed_url['pass']);
+
+        if(isset($_SERVER['HTTPS'])) {
+            if ($_SERVER['HTTPS'] != "") {
+                unset($parsed_url['port']);
+            }
+        }
+
+        // unparse URL for people without http_build_url()
+        // see http://php.net/manual/en/function.parse-url.php#106731
+        $scheme   = isset($parsed_url['scheme']) ? $parsed_url['scheme'] . '://' : ''; 
+        $host     = isset($parsed_url['host']) ? $parsed_url['host'] : ''; 
+        $port     = isset($parsed_url['port']) ? ':' . $parsed_url['port'] : ''; 
+        $user     = isset($parsed_url['user']) ? $parsed_url['user'] : ''; 
+        $pass     = isset($parsed_url['pass']) ? ':' . $parsed_url['pass']  : ''; 
+        $pass     = ($user || $pass) ? "$pass@" : ''; 
```

Comment:
user/pass stuff here is unnecessary since we always unset those properties above.